src/main/java/io/supertokens/webserver/api/multitenancy/thirdparty/RemoveThirdPartyConfigAPI.java

```diff
@@ -81,9 +82,13 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
                     config.tenantIdentifier,
                     config.emailPasswordConfig,
                     new ThirdPartyConfig(
-                            config.thirdPartyConfig.enabled, newProviders.toArray(new ThirdPartyConfig.Provider[0])),
+                            config.thirdPartyConfig.enabled,
+                            getVersionFromRequest(req).lesserThan(SemVer.v5_1) ?
```

Comment:
why checking version here? The value of useThirdPartyProvidersFromStaticConfigIfEmpty should be already set correctly based on when it was created.

Reply:
yes, but when add/update/delete thirdParty config, we should set it to false for CDI >= 5.1. otherwise, preserve the older state